registry/registry.go

```diff
@@ -61,6 +65,54 @@ func hasFile(files []os.FileInfo, name string) bool {
 	return false
 }
 
+// ReadCertsDirectory reads the directory for TLS certificates
+// including roots and certificate pairs and updates the
+// provided TLS configuration.
+func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
```

Comment:
can we use https://github.com/docker/docker/blob/master/pkg/tlsconfig/config.go#L83 to generate the TLS config rather than doing it manually here?